src/plugins/share/public/components/url_panel_content.tsx

```diff
@@ -282,6 +311,17 @@ export class UrlPanelContent extends Component<Props, State> {
     }
   };
 
+  private handleUrlParamChange = (optionId: string): void => {
+    const param = optionId as keyof UrlParams;
+    const stateUpdate: Partial<State> = {
```

Comment:
nit: If you don't declare this to be a `Partial<State>`, then you don't need the type cast in line 322.